lightningd/dual_open_control.c

```diff
@@ -1330,8 +1330,17 @@ wallet_update_channel_commit(struct lightningd *ld,
 	wallet_channel_save(ld->wallet, channel);
 
 	/* Set inflight data & update */
-	inflight_set_last_tx(inflight, remote_commit, *remote_commit_sig);
-	wallet_inflight_save(ld->wallet, inflight);
+	if (inflight->last_tx) {
+		struct bitcoin_txid txid, inflight_txid;
+		/* confirm they're the same tx! */
+		bitcoin_txid(remote_commit, &txid);
+		bitcoin_txid(inflight->last_tx, &inflight_txid);
+		/* we check validity in dualopend, this is a sanity check */
+		assert(bitcoin_txid_eq(&txid, &inflight_txid));
```

Comment:
As a rule (loosely followed!) we prefer to fail the channel if a subd does something weird, rahter than kill lightningd.